core/cmd_size.go

```diff
@@ -0,0 +1,112 @@
+package core
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/peak/s5cmd/log"
+	"github.com/peak/s5cmd/opt"
+	"github.com/peak/s5cmd/storage"
+)
+
+type sizeAndCount struct {
+	size  int64
+	count int64
+}
+
+func (s *sizeAndCount) addObject(obj *storage.Object) {
+	s.size += obj.Size
+	s.count++
+}
+
+func Size(ctx context.Context, job *Job) *JobResponse {
+	src := job.args[0]
+
+	client, err := storage.NewClient(src)
+	if err != nil {
+		return jobResponse(err)
+	}
+
+	storageTotal := map[string]sizeAndCount{}
+	total := sizeAndCount{}
+
+	for object := range client.List(ctx, src, true, storage.ListAllItems) {
+		if object.Type.IsDir() || object.Err != nil {
+			// TODO(ig): expose or log the error
+			continue
+		}
+		storageClass := string(object.StorageClass)
+		s := storageTotal[storageClass]
+		s.addObject(object)
+		storageTotal[storageClass] = s
+
+		total.addObject(object)
+	}
+
+	if !job.opts.Has(opt.GroupByClass) {
+		m := SizeMessage{
+			Source:        src.String(),
+			Count:         total.count,
+			Size:          total.size,
+			ShowHumanized: job.opts.Has(opt.HumanReadable),
+		}
+		log.Logger.Info(m)
+		return jobResponse(err)
+	}
+
+	for k, v := range storageTotal {
+		m := SizeMessage{
+			Source:        src.String(),
+			StorageClass:  k,
+			Count:         v.count,
+			Size:          v.size,
+			ShowHumanized: job.opts.Has(opt.HumanReadable),
+		}
+		log.Logger.Info(m)
+	}
+
+	return jobResponse(nil)
+}
+
+// SizeMessage is the structure for logging disk usage.
+type SizeMessage struct {
+	Source       string `json:"source"`
+	StorageClass string `json:"storage_class,omitempty"`
+	Count        int64  `json:"count"`
+	Size         int64  `json:"size"`
+
+	ShowHumanized bool `json:"-"`
```

Comment:
we can make this field private since it is internal

Reply:
Done. I also updated other internal fields. It was intended for `message` package, that's why all of the fields were public.